src/puya/awst_build/validation/arc4_copy.py

```diff
@@ -44,8 +44,3 @@ def _check_for_arc4_copy(self, expr: awst_nodes.Expression) -> None:
     def visit_assignment_expression(self, expr: awst_nodes.AssignmentExpression) -> None:
         super().visit_assignment_expression(expr)
         self._check_for_arc4_copy(expr.value)
-
-    def visit_subroutine_call_expression(self, expr: awst_nodes.SubroutineCallExpression) -> None:
```

Comment:
Wouldn't this still need to be in there, in order to check nested items, e.g. an arc4 array inside a regular tuple?

Reply:
Or, looks like maybe you got that case working? But we'd still need to check if say they passed a reference to a nested arc4.Struct or some such, right?

Reply:
```py
    @subroutine
    def check_copy_required(self) -> None:
        d_array = DynamicArray(String("Test"))

        nested_in_array = DynamicArray(d_array)

        nested_in_struct = StructWithArray(d_array=nested_in_array[0])

        assert d_array == nested_in_struct.d_array
```

Are we thinking these scenarios should require a copy? They wouldn't be under subroutine calls though

Reply:
Specifically for this PR I was thinking `some_func(nested_in_struct.d_array)` should still error.

Reply:
I'll address both